### ISSUE TITLE
sys-apps/edac-utils does not compile with LTO

### ISSUE THREAD

**Issue by jfikar** [InBetweenNames/gentooLTO#580]
sys-apps/edac-utils-0.18 fails with LTO:

```
 x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I. -I../.. -march=native -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=auto -fuse-linker-plugin -pipe -Wl,-O1 -Wl,--as-needed -c libedac.c  -fPIC -DPIC -o .libs/libedac.o
/bin/sh ../../libtool --mode=link --tag=CC x86_64-pc-linux-gnu-gcc  -march=native -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=auto -fuse-linker-plugin -pipe -Wl,-O1 -Wl,--as-needed  -Wl,-O1 -Wl,--as-needed -march=native -O3 -fgraphite-identity -floop-nest-optimize -fdevirtualize-at-ltrans -fipa-pta -fno-semantic-interposition -flto=auto -fuse-linker-plugin -pipe -o libedac.la -rpath /usr/lib64 -export-symbols-regex "^edac_.*" -version-info 2:1:1 libedac.lo -lsysfs
generating symbol list for `libedac.la'
/usr/bin/nm -B  .libs/libedac.o  |  | /bin/sed 's/.* //' | sort | uniq > .libs/libedac.exp
../../libtool: eval: line 3997: syntax error near unexpected token `|'
../../libtool: eval: line 3997: `/usr/bin/nm -B  .libs/libedac.o  |  | /bin/sed 's/.* //' | sort | uniq > .libs/libedac.exp'
make[3]: *** [Makefile:281: libedac.la] Error 2
```
adding `sys-apps/edac-utils *FLAGS-=-flto*` solves the problem

**Comment by jiblime** [comment]
So I made the mistake of spending time on this instead of real work. This is what I've found so far:
Bug report discussion https://sourceware.org/bugzilla/show_bug.cgi?id=25355
Guide on fixing it in ebuild https://wiki.gentoo.org/wiki/Outdated_libtool



**Problem:** There are two consecutive `|` `|` in the line below
`/usr/bin/nm -B  .libs/libedac.o  `**|  |**`/bin/sed 's/.* //' | sort | uniq > .libs/libedac.exp`

**Cause:** 
The value for the `global_symbol_pipe` is missing due to how `nm` works with  LTO, specifically for GCC (as I understand it)

**What should be the solution:**
Using at least GCC 10 as recommended by H.J. Lu as well as using binutils >=2.34 since those 2.34/2.35 include his fix for nm/LTO
I've personally been unable to build if even only one of them didn't meet the requirement

---

In this case, instead of that just being the solution we have one more problem reveal itself: the bundled libtool in edac-utils is from the caveman era and needs to be fixed up, else the same error will occur. I referred to the wiki above and the ebuilds it mentions.


Solutions you can try without stripping `-flto`:


**Solution 1:**
```
NOCOMMON_OVERRIDE_LIBTOOL=yes emerge edac-utils

       This will cache the missing variables for you. More information about
       what it does is in /etc/portage/bashrc.d/43-lto-no-common.sh
       I found/refound(?) this one a few minutes ago and it's actually really cool.
       The only difference between my system's libtool doesn't have the white
       spaces truncated for global_symbol_pipe.
```

**Solution 2:**
```
Rewrite the ebuild according to this guide - https://wiki.gentoo.org/wiki/Outdated_libtool
       It's a bit dated, in that it's preferable to use autoconf/autoreconf/etc in ebuilds now.
```


<details>
<summary>My ebuild here:</summary>

```
# Copyright 1999-2020 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

inherit autotools eutils flag-o-matic

DESCRIPTION="Userspace helper for Linux kernel EDAC drivers"
HOMEPAGE="https://github.com/grondo/edac-utils"
SRC_URI="https://github.com/grondo/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"

LICENSE="GPL-2"
SLOT="0"
KEYWORDS="amd64"
IUSE="debug"

DEPEND="sys-fs/sysfsutils"
RDEPEND="${DEPEND}
	sys-apps/dmidecode"

src_prepare() {
	default

	# <GCC-10, <binutils-2.34 do not support LDPT_ADD_SYMBOLS_V2 https://sourceware.org/bugzilla/show_bug.cgi?id=25355
	if tc-is-gcc && ver_test $(gcc-version) -lt 10.0 || ! tc-ld-is-lld && ver_test $(ld -v | sed 's/.*\ \(.*\)/\1/') -lt 2.34; then
		filter-flags -flto*
	fi

	sed -i \
		-e 's|-Werror||' \
		configure || die

	eautoreconf
}

src_configure() {
	econf \
		--disable-static \
		$(use_enable debug)
}

src_install() {
	default

	# We don't need this init.d file
	# Modules should be loaded by adding them to /etc/conf.d/modules
	# The rest is done via the udev-rule
	rm -rf "${D}/etc/init.d"

	find "${D}" -name '*.la' -type f -delete || die
}

pkg_postinst() {
	elog "There must be an entry for your mainboard in /etc/edac/labels.db"
	elog "in case you want nice labels in /sys/module/*_edac/"
	elog "Run the following command to check whether such an entry is already available:"
	elog "    edac-ctl --print-labels"
}
```
</details>

As `sys-config/ltoize` states, it is highly recommended that you have at least `>=sys-devel/binutils-2.34`. The fix mentioned above was patched into for 2.34 and 2.35; it's been a little over half a year since so I don't believe 2.33 will be receiving the same treatment.

**Comment by jfikar** [comment]
Good job @jiblime, I'm using the first solution. The upgraded unstable binutils 2.34 solve by itself a problem with dev-util/google-perftools.
